### ISSUE TITLE
Remove support for DIMM hotplug

### ISSUE THREAD

**Issue by stradig** [neondatabase/autoscaling#1060]
This code is no longer needed since we moved to virtio-mem on all nodes and tested this over a sufficient time. So we should remove it to reduce tech debt and clean up the code.

**Comment by sharnoff** [comment]
Update: Waiting on me to rebase #1070.

**Comment by sharnoff** [comment]
Status update: #1070 has been updated, now awaiting re-review.

**Comment by sharnoff** [comment]
Status update: #1070 has been released; #1215 is now ready for review.

**Comment by sharnoff** [comment]
With #1215 released, this is now complete
